numpyro/infer/__init__.py

```diff
@@ -29,6 +31,11 @@
 
 from . import autoguide, reparam
 
+warnings.filterwarnings(
+    "ignore", message=".*Attempting to hash a tracer.*", category=FutureWarning
+)
```

Comment:
Could we move this to `__init__` before all imports? I found that the warning still appears in some examples like `examples/gaussian_shells.py --enum`.

Reply:
Changed in https://github.com/pyro-ppl/numpyro/pull/1817/commits/0afb74689a04acebdad100027115586d800a3567

Reply:
But I think this does not fix the type errors right?

Reply:
Do you think that moving this to `numpyro/__init__.py` is better? There are other places that we import `funsor`.

Reply:
Re type error: let me take further look. I guess we just need to cast types properly.

Reply:
I tried but have not been able to fix the types :(